File: TransCoder_tokenized_test_set_functions/python/COUNT_SUBSETS_DISTINCT_EVEN_NUMBERS.py
def countSubSets ( arr, n ) :
    us = set ( )
    even_count = 0
    for i in range ( n ) :
        if arr [ i ] % 2 == 0 :
            us . add ( arr [ i ] )
    for i in us :
        even_count += 1
    return pow ( 2, even_count ) - 1

File: TransCoder_tokenized_test_set_functions/cpp/COUNT_SUBSETS_DISTINCT_EVEN_NUMBERS.cpp
int countSubsets ( int arr [ ], int n ) {
  unordered_set < int > us;
  int even_count = 0;
  for ( int i = 0;
  i < n;
  i ++ ) if ( arr [ i ] % 2 == 0 ) us . insert ( arr [ i ] );
  unordered_set < int > :: iterator itr;
  for ( itr = us . begin ( );
  itr != us . end ( );
  itr ++ ) even_count ++;
  return ( pow ( 2, even_count ) - 1 );
}